app/models/order.rb

```diff
@@ -385,6 +391,7 @@ def finalize!(sold_on_date = Time.current)
         self.items += vouchers
         self.items += retail_items
         self.items << donation if donation
```

Comment:
i'm unclear on this logic.  a single order may contain a regular donation OR a recurring donation, but (i would think) not both.  but this logic doesn't seem to account for that, and makes it look like they could both occur.  or is it the case that when the order includes a recurring donation, it is also made to include the first (regular) donation in that series?  either way, the logic here isn't clear and i'd like to go over this during our next/final client meeting.

Reply:
> or is it the case that when the order includes a recurring donation, it is also made to include the first (regular) donation in that series?

This is correct: when the order includes a recurring donation, it also includes the first donation charge in that series.

Currently, the controller action that processes donations made from the /donate page instantiates an order that contains only a regular donation. Our idea is that in the case of a one-time donation, this simply represents the one-time charge, and in the case of a recurring donation, this represents the first charge in the series. With this design, the donation object is always recorded with the order, and the new logic simply adds a recurring donation to the order when appropriate (to record that a series of related donations has begun).